Task 10

pavel-kostoglod opened the pull request:
- RealItemTest with: 1 test
- VirtualItemTest with: 1 test
- JsonParserTest with: 2 tests, @Disabled annotation, @BeforeEach and @AfterEach annotations

Change-Id: I6aba00f254be712f08bb477ca175b781cf036a31

Commit:
- CartTest with: 2 tests, @BeforeAll annotation
- RealItemTest with: 1 test
- VirtualItemTest with: 1 test
- JsonParserTest with: 2 tests, @Disabled annotation, @BeforeEach and @AfterEach annotations

Change-Id: I6aba00f254be712f08bb477ca175b781cf036a31

KirKarpov commented on src/test/java/parser/JsonParserTest.java:
```diff
@@ -0,0 +1,52 @@
+
+import java.io.File;
+
+public class JsonParserTest {
```
I don't see tests that check how works writeToFile and readFromFile methods.
You will need to write your own readFromFile method to validate work on writeToFile and vice versa for readFromFile method

KirKarpov commented on src/test/java/shop/CartTest.java:
```diff
@@ -0,0 +1,50 @@
+    }
+
+    @Test
+    void getTotalPriceEqualToSumOfItemsPricesPlusTax() {
```
You checked how price will be calculated when you adding realItem and virtualItem. You also need to check calculation when you remove it.

pavel-kostoglod committed:
- checkTotalPriceAfterItemsRemoving method added to the Cart checks (Is there a mistake in the code? Cause there is no decrease of "TotalSum" after removing item from the cart)

- writeToFileWritesCorrectCartInfo and readFromFileReadsCorrectCartInfo methods added to the JsonParser checks. It's not the easiest way, but Reflection seems convenient in that case.

pavel-kostoglod commented:
Changes added

KirKarpov commented on src/test/java/parser/JsonParserTest.java:
```diff
@@ -0,0 +1,125 @@
+
+    @Test
+    void writeToFileWritesCorrectCartInfo() {
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
```
It is better to move you logic related to reading the file to some util class. It could be named as Utilities
It will allow to reuse the code in other tests and remove code overload from test

KirKarpov commented on src/test/java/parser/JsonParserTest.java:
```diff
@@ -0,0 +1,125 @@
+    }
+
+    @Test
+    void writeToFileWritesCorrectCartInfo() {
```
This method is not writing to file. It is reading from file

KirKarpov commented on src/test/java/parser/JsonParserTest.java:
```diff
@@ -0,0 +1,125 @@
+    }
+
+    @Test
+    void readFromFileReadsCorrectCartInfo() {
```
As an option you can use reflection to read file :)
The first option (with BufferedReader ) looks a bit easier. 
But the same code of reading the file should me moved to utility class

KirKarpov commented on src/test/java/shop/CartTest.java:
```diff
@@ -0,0 +1,61 @@
+    void checkTotalPriceAfterItemsRemoving() {
+        cart.deleteRealItem(realItem);
+        cart.deleteVirtualItem(virtualItem);
+        Assertions.assertEquals((
+                (realItemPrice + realItemPrice*TAX) +
+                (virtualItemPrice + virtualItemPrice*TAX)),
+                cart.getTotalPrice()
+        );
```
if you are removing items should the price be calculated based on items that was removed. In your case, when you removed all items, I suppose the price should 0

pavel-kostoglod replied:
That was my question. No total sum decreasing in the delete item method

pavel-kostoglod committed:
- Changed the logic of Cart test after items removing
- Changed the logic of JsonParser tests for write to file and read from file
- Common methods moved to utils package
